src/js/module.js

```diff
@@ -8,3 +8,72 @@ Module.noImageDecoding = true;
 Module.noAudioDecoding = true;
 Module.noWasmDecoding = false; // we preload wasm using the built in plugin now
 Module.preloadedWasm = {};
+
+/**
+ *
+ * @param {undefined|(() => string)} stdin
+ * @param {undefined|((text: string) => void)} stdout
+ * @param {undefined|((text: string) => void)} stderr
+ */
+export function setStandardStreams(stdin, stdout, stderr) {
+  if (stdout) {
+    Module.print = stdout;
+  }
+
+  if (stderr) {
+    Module.printErr = stderr;
+  }
+
+  if (stdin) {
+    Module.preRun = [
+      function () {
+        Module.FS.init(createStdinWrapper(stdin), null, null);
```

Comment:
Could we have a comment explaining why we set `stdin` with `FS.init` but set `stdout` and `stderr` with `Module.print` and `Module.printErr`?

Reply:
Sure, I added a comment. It's basically because that's how the Emscripten API is.

Reply:
Perfect.